src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,79 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Deals with loading tasks from the file and saving tasks in the file
+ */
+public class Storage {
+    private final List<Task> tasks = new ArrayList<>();
+    private final Parser parser = new Parser();
+
+    /**
+     * Loads the task list information from the hard disk.
+     *
+     * @return A list of tasks retrieved from the hard disk.
+     */
+    public List<Task> loadFromDisk() {
+        File folder = new File("data");
+        if (!folder.isDirectory()) {
+            createDataFolder();
+            createFile();
+        }
+
+        try {
+            File file = new File("data/duke.txt");
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNext()) {
+                String taskString = scanner.nextLine();
+                Task task = parser.taskStringToTask(taskString);
+                this.tasks.add(task);
+            }
+        } catch (FileNotFoundException e) {
+            //create the file if it does not exist.
+            createFile();
+        }
+        return tasks;
+    }
+
+    /**
+     * Saves the list changes on the hard disk.
+     */
+    public void saveProgress() {
+        try {
+            FileWriter fw = new FileWriter("data/duke.txt");
+            for (Task task : tasks) {
+                fw.write(task + "\n");
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("No File Found");
+            System.exit(-1);
+        }
```

Comment:
Shouldn't adding a finally block to close the Filewriter instead of closing it inside the try block better ? (In the case of an exception occuring in the try code).